index.js

```diff
@@ -92,6 +125,11 @@ function Metric(namespace, units, defaultDimensions, options) {
   this.units = units;
   this.defaultDimensions = defaultDimensions || [];
   this.options = _.defaults(options || {}, DEFAULT_METRIC_OPTIONS);
+  this._storedMetrics = [];
+
+  if (this.options.enabled) {
+    setInterval(this._sendMetrics, this.options.sendInterval);
```

Comment:
So, are analytics bursty? If so it might be better to not call this on an interval, because you could have the first item of the burst "make it" right before `_sendMetrics` was called and the rest wait 5 seconds, but instead wait to schedule a `_sendMetrics` call until `put` was called, i.e. use `_.throttle` with `{leading: false}`.